mycroft/util/parse.py

```diff
@@ -164,6 +169,571 @@ def extractnumber_en(text):
     return val
 
 
+'''authors: spenrod and connorpenrod
+this function takes a sentence and extracts date and time information from it
+returns a 2-element array [date, leftover_string]
+the 'date' is a python datetime object
+the 'leftover_string' contains all keywords not related to the date or time'''
+
+
```

Comment:
Don't use currentDate=datetime.now().  Python is a little odd in the way it handles default values.  It only gets executed once in the environment.  Try this example in the interactive interpreter:

```
> import datetime

> def func(paramVal=datetime.datetime.now()):
...    print paramVal
```
Now run func() several times.  You'll be surprised.


Instead, do this:

```
   def extract_datetime_en(str, currentDate=None):
       if currentDate is None:
           currentDate = datetime.now()
       ....
```